### PR TITLE
JBPM-9820 Navigation link / (+) button not working when sub-process c…

### PR DESCRIPTION
**Pull request by bxf12315**
…ontains boundary event

**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[https://issues.redhat.com/browse/JBPM-9820](JBPM-9820)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Review by jstastny-cz — COMMENTED** [review]
I am not sure I follow what and how we're fixing in these PRs, could you please summarize?

**Comment by bxf12315** [comment]
Jenkins run fdb

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/StunnerSVGProcessor.java [hunk 1]
```diff
@@ -100,8 +100,10 @@ private void processNode(final Node parent, final String nodeId) {
                         if (subProcessLinks != null && subProcessLinks.containsKey(nodeId) && (value.equals(nodeId + SPECIAL_SUB_PROCESS_INSTANCE_ID) || value.equals(nodeId + OLD_SPECIAL_SUB_PROCESS_INSTANCE_ID))) {
                             plusButton = (Element) node;
                         }
-
-                        summary.addNode(new NodeSummary(nodeId, border, background, null, null, renderType, plusButton));
+                        /** RHPAM-3779 **/
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/StunnerSVGProcessor.java]: updated

**Comment by bxf12315** [comment]
@nmirasch UT added.

**Review by nmirasch — APPROVED** [review]

File: jbpm-process-svg/src/test/java/org/jbpm/process/svg/TestEvalutionSVG.java [hunk 2]
```diff
@@ -109,26 +109,33 @@ public void testSubProcessLink() throws Exception {
 
     @Test
     public void testStunnerSubProcessLink() throws Exception {
-        String elementId = "_D254259B-8C23-4498-9BE1-995E2DC66726";
+        testSubProcessLink("_D254259B-8C23-4498-9BE1-995E2DC66726", "/stunner.parentP-svg.svg");
+    }
+
+    @Test
+    public void testBoundaryEventSubProcessLink() throws Exception {
+        testSubProcessLink("_86E33F6B-FD3B-4E07-8D60-D273A54FE6B7", "/svg-test.svg");
```
Comment (jstastny-cz) [inline jbpm-process-svg/src/test/java/org/jbpm/process/svg/TestEvalutionSVG.java:117]: please rename the test file to indicate purpose - see the file above is `stunner.parentP-svg.svg`, could this be `stunner.parentP.boundaryEvent-svg.svg` ?

Comment (bxf12315) [reply jbpm-process-svg/src/test/java/org/jbpm/process/svg/TestEvalutionSVG.java]: done

**Comment by mareknovotny** [comment]
jenkins retest this please

**Comment by jstastny-cz** [comment]
Using the provided example in Jira this does not work properly I am afraid.

I did reach following state 
Screenshot from 2021-07-21 18-22-16

* A was completed by sending signal to the child - it's still clickable from the parent after completion.
* B was aborted by sending signal to the parent (caught by boundary event) - the child is not clickable anymore, even though it was before sending the event.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]
Confirmed, once the boundary event aborts the child process it is not clickable anymore, as oposed to any completed child.

When child completes:
Screenshot from 2021-07-21 19-25-58
* Link navigation still works

When child is aborted (using the sub-process boundary event in parent):
Screenshot from 2021-07-21 19-26-11
* plus sign is not clickable anymore (and svg path does not have the onclick attribute)

**Review by jstastny-cz — APPROVED** [review]
After off-line discussion, approving this change with the reported corner-case to be tracked as a separate follow-up issue.

**Pull request merged**